packages/generator/templates/page/__id__.tsx

```diff
@@ -6,23 +6,46 @@ import delete__ModelName__ from 'app/__modelNames__/mutations/delete__ModelName_
 export const __ModelName__ = () => {
   const router = useRouter()
   const id = parseInt(router?.query.id as string)
+  if (process.env.parentModel) {
+    const __parentModelId__ = parseInt(router?.query.__parentModelId__ as string)
+  }
   const [__modelName__] = useQuery(get__ModelName__, {where: {id}})
 
+  const editLink = process.env.parentModel
+    ? (
+      <Link
+        href="/__parentModels__/__parentModelParam__/__modelNames__/[id]/edit"
+        as={`/__parentModels__/${__parentModelId__}/__modelNames__/${__modelName__.id}/edit`}
+      >
+        <a>Edit</a>
+      </Link>
+    )
+    : (
+      <Link href="/__modelNames__/[id]/edit" as={`/__modelNames__/${__modelName__.id}/edit`}>
+        <a>Edit</a>
+      </Link>
+    )
+
   return (
     <div>
       <h1>__ModelName__ {__modelName__.id}</h1>
       <pre>{JSON.stringify(__modelName__)}</pre>
 
-      <Link href="/__modelNames__/[id]/edit" as={`/__modelNames__/${__modelName__.id}/edit`}>
-        <a>Edit</a>
-      </Link>
+      {editLink}
```

Comment:
Don't have this before merging this, but it'd be really nice if we can somehow support having conditional components here without the separate `editLink` component. Because this results in a weird output in the generated file (a separate `editLink` component for no reason)

Reply:
yeah, I didn't love this, the problem is it's a limitation of the JSX syntax (can't put JS in JSX without `{}`). We could eventually build some handling for jsx conditionals with something like this:

```xml
<if condition="someTemplateValue">
 <Link />
  <else>
    <OtherLink />
  </else>
</if>
```

but that's a pretty significant change that we should split into another issue